### PR TITLE
Make non-global RegExps global so they can "replace all" occurrences

### PR DESCRIPTION
**Pull request by mathiasbynens**
This is one of the potential solutions to #16.

### CODE REVIEW

**Review by ExE-Boss — APPROVED** [review]
I am in favour of this.

**Comment by mathiasbynens** [comment]
Alternative approach: #24

File: spec.html [hunk 1]
```diff
@@ -77,3 +77,19 @@ <h1>String.prototype.replaceAll ( _searchValue_, _replaceValue_ )</h1>
...
   </emu-alg>
 </emu-clause>
+
+<emu-clause id="sec-regexp.prototype-@@replaceall">
+  <h1>RegExp.prototype [ @@replaceAll ] ( _string_, _replaceValue_ )</h1>
+  <p>When the `@@replaceAll` method is called with arguments _string_ and _replaceValue_, the following steps are taken:</p>
+  <emu-alg>
+    1. Let _rx_ be the *this* value.
+    1. If Type(_rx_) is not Object, throw a *TypeError* exception.
+    1. Let _flags_ be ? ToString(? Get(_rx_, `"flags"`)).
```
Comment (ljharb) [inline spec.html:87]: if flags is undefined, this should be the empty string, not the string undefined, to match the RegExp constructor.

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by mathiasbynens** [comment]
We're going with #24 instead.

**Pull request closed without merging**